hal/src/main/java/edu/wpi/first/hal/simulation/SimulatorJNI.java

```diff
@@ -21,4 +22,48 @@
   public static native void stepTiming(long delta);
   public static native void stepTimingAsync(long delta);
   public static native void resetHandles();
+
+  public static class SimPeriodicBeforeCallback implements AutoCloseable {
+    public SimPeriodicBeforeCallback(Runnable r) {
```

Comment:
Should these constructors be public? It seems like these are meant to be instantiated only by the register static methods.

Reply:
Changed them to private.